modules/medianetBidAdapter.js

```diff
@@ -363,38 +368,78 @@ function fetchCookieSyncUrls(response) {
   return [];
 }
 
-function getLoggingData(event, data) {
-  data = (isArray(data) && data) || [];
-
-  let params = {};
+function getEventData(event) {
+  const params = {};
+  const referrerInfo = getRefererInfo();
   params.logid = 'kfk';
   params.evtid = 'projectevents';
   params.project = 'prebid';
-  params.acid = deepAccess(data, '0.auctionId') || '';
+  params.pbver = '$prebid.version$';
   params.cid = getGlobal().medianetGlobals.cid || '';
-  params.crid = data.map((adunit) => deepAccess(adunit, 'params.0.crid') || adunit.adUnitCode).join('|');
-  params.adunit_count = data.length || 0;
-  params.dn = mnData.urlData.domain || '';
-  params.requrl = mnData.urlData.page || '';
-  params.istop = mnData.urlData.isTop || '';
+  params.dn = encodeURIComponent(referrerInfo.domain || '');
+  params.requrl = encodeURIComponent(referrerInfo.page || '');
   params.event = event.name || '';
   params.value = event.value || '';
   params.rd = event.related_data || '';
+  return params;
+}
+
+function getBidData(bid) {
+  const params = {};
+  params.acid = bid.auctionId || '';
+  params.crid = deepAccess(bid, 'params.crid') || deepAccess(bid, 'params.0.crid') || bid.adUnitCode || '';
+  params.ext = safeJSONEncode(bid.ext) || '';
 
+  const rawobj = deepClone(bid);
+  delete rawobj.ad;
+  delete rawobj.vastXml;
+  params.rawobj = safeJSONEncode(rawobj);
   return params;
 }
 
-function logEvent (event, data) {
-  let getParams = {
-    protocol: 'https',
-    hostname: EVENT_PIXEL_URL,
-    search: getLoggingData(event, data)
-  };
-  triggerPixel(buildUrl(getParams));
+function getLoggingData(event, bids) {
+  const logData = {};
+  bids.forEach?.((bid) => {
+    let bidData = getBidData(bid);
+    Object.keys(bidData).forEach((key) => {
+      logData[key] = logData[key] || [];
+      logData[key].push(encodeURIComponent(bidData[key]));
+    });
+  });
+  return Object.assign({}, getEventData(event), logData)
+}
+
+function firePostLog(url, payload) {
+  try {
+    const isSent = window.navigator.sendBeacon(url, payload);
```

Comment:
We require our methods for network traffic

Reply:
We have removed sendBeacon() for now since Prebid doesn't currently support it. We suggest adding sendBeacon() to Prebid, as it is the most reliable for analytics use cases, as mentioned in https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon .

Reply:
fetch keepalive is better :) but firefox is still working on adding support